core-build/gulp-core-build-serve/src/ServeTask.ts

```diff
@@ -102,8 +102,62 @@ export class ServeTask<TExtendedConfig = {}> extends GulpTask<IServeTaskConfig &
   }
 
   public executeTask(gulp: typeof Gulp, completeCallback?: (error?: string) => void): void {
+    /*
+     * Workaround for loading gulp-connect, which automatically uses http2 if it
+     * can require() a module called 'http2'
+     *
+     * In versions of NodeJS < 8:
+     *   'http2' would have to be a module in the node_modules folder.
+     *   We did not provide this normally, so most of the time nobody used http2 with gulp serve.
+     *
+     * However, in versions of NodeJS >= 8:
+     *   They provide a built-in module called 'http2', which is experimental and unstable
+     *   The built in module is preferred unless the an environment variable is set: NODE_NO_HTTP2=1
+     *
+     * We don't want people to have to change environment variables to run our stuff, nor
+     * do we want people on Node8 using an experimental version of http2 that may break.
+     *
+     * Therefore, we will insert a dummy export into the require cache which is undefined.
+     * This will trick gulp-connect into thinking that HTTP2 is never available, which will make
+     * things work as they always did.
+     *
+     * However, this will  using Node < 8, who also had a module called 'http2' in their
+     * node_modules
```

Comment:
As a consequence, this approach will prevent "gulp-connect" from using http2 in environments with Node < 8. This is intentional, because we don't see a reason to support http2 for serving localhost scripts.